### PR TITLE
feat: #999 - RTK consolidate reducers

### PR DESCRIPTION
**Pull request by MarioAslau**
## **Related issues**

Fixes: #

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

**Issue**: https://github.com/MetaMask/mobile-planning/issues/999

**Description**

The purpose of this PR is to migrate our store from old redux to  [Redux Tool Kit](https://redux-toolkit.js.org/) - by focusing solely on updating the engine

**Changes**:
- installed @reduxjs/toolkit
- updated store to use configureStore
- created file persistConfig to export the persistedConfig object to the store file which will be more focused on configuring the store itself only
- created a new space for future redux files in redux folder: for selectors and slices
- migrated engine reducer into a state slice
- added unit tests for engine slice

**E2E Test**: 
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/b2781b20-539c-4f9e-b987-f9678ebc4ca5

**Possible future improvements:**
- finish migrating reducers to RTK - ones that are mutating the state correctly can use createSlice or createReducer !
- replace thunk with createAsyncThunk or [React Query](https://redux-toolkit.js.org/rtk-query/overview)
- move store file in reduxLayer 
- Upgrade Redux-Saga

**NOTION Page for the findings**: https://www.notion.so/Redux-Toolkit-Performance-Upgrade-b10cb97d7ed44dfaa15b522951971037

**Current Reducers that need refactoring in the future for reference (out of scope):**

- collectiblesReducer
- privacyReducer
- bookmarksReducer
- browserReducer
- modalsReducer
- settingsReducer
- alertReducer
- transactionReducer
- userReducer
- wizardReducer
- onboardingReducer
- notificationReducer
- swapsReducer
- fiatOrders
- infuraAvailabilityReducer
- navigationReducer
- networkOnboardReducer
- securityReducer

### CODE REVIEW

**Comment by MarioAslau** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by jpcloureiro — DISMISSED** [review]
looks good to me!

File: app/core/EngineService/EngineService.ts [hunk 1]
```diff
@@ -85,15 +84,22 @@ class EngineService {
 
     engine?.datamodel?.subscribe?.(() => {
       if (!this.engineInitialized) {
-        store.dispatch({ type: INIT_BG_STATE_KEY });
+        store.dispatch(
+          actions.initializeEngineState(UntypedEngine.state as any),
```
Comment (frankvonhoven) [inline app/core/EngineService/EngineService.ts:88]: Maybe I don't have enough context, but couldn't this be typed as `EngineState` as exported [here](https://github.com/MetaMask/metamask-mobile/blob/da85a9b1925b0615b2e967442e1a5ff053524360/app/core/Engine.ts#L141C18-L141C29)?

Comment (MarioAslau) [reply app/core/EngineService/EngineService.ts:88]: I tried doing that initially, went down a rabbit hole to fix Engine types. Eventually decided to use any
Screenshot 2023-09-19 at 16 56 51

**Review by jpcloureiro — DISMISSED** [review]

**Comment by MarioAslau** [comment]
BUG: "You don't have any tokens" - 
Screenshot 2023-09-20 at 23 24 31

Have been investigating what is causing this bug.

Bug: When the app does a fresh install and import of a wallet, the first time the app runs this screen will turn up. On any kind of app refresh, the state will update accordingly.

**Things tested so far:**

- RTK store implementation works with old reducer
- RTK store implementation works with a new reducer that used createAction and createReducer hooks
- currently the bug appears when using createSlice, update engine state reducer does not get called anymore the second time to update the state

**Review by Cal-L — DISMISSED** [review]
LGTM

**Review by Andepande — APPROVED** [review]
QA-Passed

**Pull request merged**